### PR TITLE
[torch] Fix aten.squeeze lowering to use result shape

### PR DESCRIPTION
**Pull request by rsuderman**
Squeezes can be ambiguous without the output shape information. For instance (1, 1, 256) squeezed can be either (1, 256) or (256). We need to check the resulting shape to know what the shape should look like.

### CODE REVIEW

**Review by zjgarvey — APPROVED** [review]
Cool, these are nice changes. I assume from context that reassociation[i] is the like the start and end dims of the input tensor that get collapsed to output at dim[i], correct?

**Comment by rsuderman** [comment]
> Cool, these are nice changes. I assume from context that reassociation[i] is the like the start and end dims of the input tensor that get collapsed to output at dim[i], correct?

Pretty much - it basically specifies which input dims collapse into which output dims. Its a little silly because they are required to be 0 ... N-1 in order so the format is a little excessive but good enough.

File: projects/pt1/python/torch_mlir_e2e_test/configs/torchdynamo.py [hunk 1]
```diff
@@ -130,16 +130,17 @@ def my_aot_autograd_backend(gm: torch.fx.GraphModule,
...
                               decompositions=_get_decomposition_table)
-
     with torch.no_grad():
         set_model_name(model.__class__.__name__)
         torch._dynamo.reset()
         dynamo_f = dynamo.optimize(my_backend, nopython=True)(
             lambda method, *inputs: method(*inputs))
         dynamo_f(lambda *inputs: model(*[x.clone() for x in inputs]),
                  *example_args)
+        print(mlir_module)
```
Comment (renxida) [inline projects/pt1/python/torch_mlir_e2e_test/configs/torchdynamo.py:140]: 1. possible debug statement left in
2. this is duplicated around something that apparently does not change the mlir module

**Review by renxida — APPROVED** [review]

File: projects/pt1/python/torch_mlir_e2e_test/configs/torchdynamo.py (hunk 1 above)
Comment (rsuderman) [reply projects/pt1/python/torch_mlir_e2e_test/configs/torchdynamo.py:140]: Debug statement that leaked through. I'll remove.
